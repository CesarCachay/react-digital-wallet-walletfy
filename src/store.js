import { createStore, applyMiddleware, compose } from "redux";
import logger from "redux-logger";
import reducer from "./reducer";

const composer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(reducer, composer(applyMiddleware(logger)));

export default store;
