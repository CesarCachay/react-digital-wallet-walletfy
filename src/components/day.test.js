import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../store";

import Day from "./day";

const date = "2019-04-01T05:01:00.000Z";

test("Day render as expense", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Day date={date} isExpense={true} />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

test("Day render as income", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Day date={date} isExpense={false} />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
