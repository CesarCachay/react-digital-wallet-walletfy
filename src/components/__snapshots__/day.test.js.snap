// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day render as expense 1`] = `
<DocumentFragment>
  .emotion-2 {
  margin: 0.5em;
  box-shadow: 0 3px 6px #e0e0dc;
}

.emotion-0 {
  background-color: #f4a0a8;
  padding: 0.25em;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #f4a0a86e;
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-0"
    >
      <span>
        Mon Apr 01 2019
      </span>
    </div>
    <div
      class="emotion-1"
    />
  </div>
</DocumentFragment>
`;

exports[`Day render as income 1`] = `
<DocumentFragment>
  .emotion-2 {
  margin: 0.5em;
  box-shadow: 0 3px 6px #e0e0dc;
}

.emotion-0 {
  background-color: #87caa657;
  padding: 0.25em;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #87caa62b;
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-0"
    >
      <span>
        Mon Apr 01 2019
      </span>
    </div>
    <div
      class="emotion-1"
    />
  </div>
</DocumentFragment>
`;
